File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0

        for i in range(1, len(prices)):
            # 가격이 상승한 날이면 그 차익을 얻는다
            if prices[i] > prices[i - 1]:
                profit += prices[i] - prices[i - 1]

        return profit

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.c
int maxProfit(int* prices, int pricesSize) {
    int profit = 0;

    for (int i = 1; i < pricesSize; i++) {
        // 전날보다 오늘 가격이 올랐다면 그 차익을 얻는다
        if (prices[i] > prices[i - 1]) {
            profit += prices[i] - prices[i - 1];
        }
    }

    return profit;
}